_query-dsl/specialized/rank-feature.md

```diff
@@ -0,0 +1,547 @@
+{ "index": { "_id": 6 } }
+{ "title": "Gaming Laptop", "popularity": 250 }
+{ "index": { "_id": 7 } }
+{ "title": "4K Monitor", "popularity": 500 }
+```
+{% include copy-curl.html %}
+
+## Basic rank feature query
+
+You can boost results based on the `popularity` score using `rank_feature`:
+
+```json
+POST /products/_search
+{
+  "query": {
+    "rank_feature": {
+      "field": "popularity"
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+This query alone does not perform filtering, rather it scores all documents based on the value of `popularity`. Higher values yield higher scores:
+
+```json
+{
+  ...
+  "hits": {
+    "total": {
+      "value": 7,
+      "relation": "eq"
+    },
+    "max_score": 0.9252834,
+    "hits": [
+      {
+        "_index": "products",
+        "_id": "7",
+        "_score": 0.9252834,
+        "_source": {
+          "title": "4K Monitor",
+          "popularity": 500
+        }
+      },
+      {
+        "_index": "products",
+        "_id": "6",
+        "_score": 0.86095566,
+        "_source": {
+          "title": "Gaming Laptop",
+          "popularity": 250
+        }
+      },
+      {
+        "_index": "products",
+        "_id": "5",
+        "_score": 0.71237755,
+        "_source": {
+          "title": "Noise Cancelling Headphones",
+          "popularity": 100
+        }
+      },
+      {
+        "_index": "products",
+        "_id": "4",
+        "_score": 0.5532503,
+        "_source": {
+          "title": "Smartwatch",
+          "popularity": 50
+        }
+      },
+      {
+        "_index": "products",
+        "_id": "3",
+        "_score": 0.38240916,
+        "_source": {
+          "title": "Portable Charger",
+          "popularity": 25
+        }
+      },
+      {
+        "_index": "products",
+        "_id": "2",
+        "_score": 0.19851118,
+        "_source": {
+          "title": "Bluetooth Speaker",
+          "popularity": 10
+        }
+      },
+      {
+        "_index": "products",
+        "_id": "1",
+        "_score": 0.024169207,
+        "_source": {
+          "title": "Wireless Earbuds",
+          "popularity": 1
+        }
+      }
+    ]
+  }
+}
+```
+
+## Combine with full-text search
+
+To filter relevant results and boost them based on popularity use the following request:
+
+```json
+POST /products/_search
+{
+  "query": {
+    "bool": {
+      "must": {
+        "match": {
+          "title": "headphones"
+        }
+      },
+      "should": {
+        "rank_feature": {
+          "field": "popularity"
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+This ranks all documents matching "headphones" and boosts those with higher popularity.
+
+## Boost parameter
+
+The `boost` parameter allows you to scale the score contribution of the rank_feature clause. It's especially useful in compound queries such as bool, where you want to control the influence of a feature relative to other conditions.
```

Comment:
> where you want to control the influence of a feature relative to other conditions.

I think we using lot of words which are not defined in the context of this page e,g; `feature`. If this can be translated into more relatable name in the context of this page it would be great otherwise lets add some brief explanation for the domain specific words.